contribute/style-guides/e2e.md

```diff
@@ -58,11 +58,13 @@ Now that we have a `Page` called `Login` in our `Pages` const we can use that to
 
 The last step in our example is to use our `Login` page as part of a test. The `pageFactory` function we used before gives us two things: 
 - The `url` property is used whenever we call the `visit` function and is equivalent to the Cypress function [cy.visit()](https://docs.cypress.io/api/commands/visit.html#Syntax).
+> Best practices after calling visit is to always call `should` on a selector to prevent flaky tests where you try to operate on a selector that isn't ready. Read more about it [here](https://docs.cypress.io/guides/core-concepts/retry-ability.html#Commands-vs-assertions).  
 - Any defined selector in the `selectors` property can be accessed from the `Login` page by invoking it and this is equivalent to the result of the Cypress function [cy.get(...)](https://docs.cypress.io/api/commands/get.html#Syntax).
 ```ecmascript 6
 describe('Login test', () => {
   it('Should pass', () => {
     e2e.pages.Login.visit();
+    e2e.pages.Login.username().should('be.visible'); // To prevent flaky tests always do a .should on an selector that you expect to be in the DOM. Read more here: https://docs.cypress.io/guides/core-concepts/retry-ability.html#Commands-vs-assertions
```

Comment:
Maybe add that as a block comment above this line for readability

Reply:
Good suggestion!